sydent/util/emailutils.py

```diff
@@ -64,16 +64,29 @@ def sendEmail(
     )
 
     allSubstitutions = {}
+
     for k, v in substitutions.items():
         allSubstitutions[k] = v
-        allSubstitutions[k + "_forhtml"] = escape(v)
-        allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
+        if not templateFile.endswith(".j2"):
+            allSubstitutions[k + "_forhtml"] = escape(v)
+            allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
 
     # We add randomize the multipart boundary to stop user input from
     # conflicting with it.
     allSubstitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(32)
 
-    mailString = open(templateFile).read() % allSubstitutions
+    # extract branded template name from templateFile
+    file_parts = templateFile.split("/")
+    template_name = file_parts[-2] + "/" + file_parts[-1]
```

Comment:
This all feels a bit weird to me -- we seem to be bouncing back and forth (unnecessarily) between the full filename and the template name.

Can we pass in the template name to this function and have it call `get_branded_template` when necessary? I think that could be used to clean things up a bit?

Or maybe if a matching `.j2` file is found in `get_branded_template` we should return *just* `template_name + ".j2"` instead of a full path to it?